src/modules/claims/claims.service.ts

```diff
@@ -790,28 +803,23 @@ export class ClaimsService {
     // add scenario for offchain without request based on claimType instead of token
     // can we break API so that register on chain required only claim type and claim type version and subject
     if (registrationTypes.includes(RegistrationTypes.OffChain)) {
-      const token = claim.token as string;
-      const payload = (await this._didRegistry.decodeJWTToken({ token })) as {
-        iss: string;
-        sub: string;
-
-        claimData: ClaimData;
-      };
-      const { iss, claimData } = payload;
-      let sub = payload.sub;
-      // Initialy subject was ignored because it was requester
-      if (!sub || sub.length === 0 || !isValidDID(sub)) {
-        sub = this._signerService.did;
+      if (!claim.token) {
+        throw new Error(ERROR_MESSAGES.CLAIM_DOES_NOT_CONTAIN_TOKEN);
       }
-      const verifiedDid = await this._didRegistry.verifyPublicClaim(token, iss);
-      if (!verifiedDid || !compareDID(verifiedDid, iss)) {
+      this._didRegistry.isClaim(payload);
```

Comment:
I think it is supposed to be used like this
```suggestion
      if (!this._didRegistry.isClaim(payload)) {
         throw new Error(...)
       };
```